### PR TITLE
[vshare.io] Add new extractor

### PR DESCRIPTION
**Pull request by piotr5151**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [x] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

### CODE REVIEW

**Comment by piotr5151** [comment]
New extractor for vshare.io

File: youtube_dl/extractor/extractors.py [hunk 1]
```diff
@@ -288,7 +288,6 @@
     ESPNArticleIE,
 )
 from .esri import EsriVideoIE
-from .etonline import ETOnlineIE
```
Comment (dstftw) [inline youtube_dl/extractor/extractors.py]: Do not remove any code.

File: youtube_dl/extractor/vshare.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<div id="root-container">\s*<div id[^>]+>\s*(.+?)<br/>\s*', webpage, 'title')
+        video_url = self._search_regex(r'<a style="text-decoration:none;" href="(.+?)">Click here</a>', webpage, 'video url')
+
+        from ..utils import determine_ext
+        ext = determine_ext(video_url)
```
Comment (dstftw) [inline youtube_dl/extractor/vshare.py]: This is determined automatically.

**Comment by dstftw** [comment]
This looks like copyrighted content. youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free). Do they have rights to distribute this? If not provide [an example URL](https://github.com/rg3/youtube-dl/blob/master/README.md#is-the-description-of-the-issue-itself-sufficient) that does not violate any copyrights.

**Comment by piotr5151** [comment]
All fixes implemented.

**Comment by dstftw** [comment]
Squash commits.

File: youtube_dl/extractor/vshare.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+class VShareIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?vshare\.io/[^/]*/(?P<id>.+)'
+    _TEST = {
+        'url': 'https://vshare.io/d/0f64ce6',
+        'md5': '16d7b8fef58846db47419199ff1ab3e7',
+        'info_dict': {
+            'id': '0f64ce6',
+            'title': 'vl14062007715967',
+            'url': r're:https?://s2\.vshare\.io/download,9999999999999999999999999999999999999999-[a-z0-9]{32},[a-z0-9]{8},0f64ce6.mp4',
```
Comment (dstftw) [inline youtube_dl/extractor/vshare.py]: No point checking this.

File: youtube_dl/extractor/vshare.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+            'ext': 'mp4',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<div id="root-container">\s*<div id[^>]+>\s*(.+?)<br/>\s*', webpage, 'title')
```
Comment (dstftw) [inline youtube_dl/extractor/vshare.py]: `\s*` make no sense at the end.

File: youtube_dl/extractor/vshare.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class VShareIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?vshare\.io/[^/]*/(?P<id>.+)'
```
Comment (dstftw) [inline youtube_dl/extractor/vshare.py]: This matches https://vshare.io//0f64ce6 that is incorrect.

**Comment by piotr5151** [comment]
dstftw, thanks for your checking. All your suggestion are implemented.

**Comment by piotr5151** [comment]
Any chance to include vshare.io into master branch ?

**Comment by piotr5151** [comment]
rebased to solve conflicts and squash commit

File: youtube_dl/extractor/vshare.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+class VShareIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?vshare\.io/[vd]/(?P<id>.+)'
+    _TEST = {
+        'url': 'https://vshare.io/d/0f64ce6',
+        'md5': '16d7b8fef58846db47419199ff1ab3e7',
+        'info_dict': {
+            'id': '0f64ce6',
+            'title': 'vl14062007715967',
+            'url': r're:https?://s\d+\.vshare\.io/download,9999999999999999999999999999999999999999-[a-z0-9]{32},[a-z0-9]{8},0f64ce6.mp4',
```
Comment (dstftw) [inline youtube_dl/extractor/vshare.py]: There is no point checking `url`.

File: youtube_dl/extractor/vshare.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+            'ext': 'mp4',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+
+        title = self._html_search_regex(r'<div id="root-container">\s*<div id[^>]+>\s*(.+?)<br/>', webpage, 'title')
+        video_url = self._search_regex(r'<a style="text-decoration:none;" href="(.+?)">Click here</a>', webpage, 'video url')
```
Comment (dstftw) [inline youtube_dl/extractor/vshare.py]: This should be more relaxed.

**Comment by dstftw** [comment]
Does not work for v-URLs: https://vshare.io/v/0f64ce6/width-650/height-430/1

**Comment by piotr5151** [comment]
Done

**Comment by dstftw** [comment]
Still not.

**Comment by piotr5151** [comment]
any hint what should be updated yet ?

**Comment by dstftw** [comment]
>Does not work for v-URLs: https://vshare.io/v/0f64ce6/width-650/height-430/1

**Comment by piotr5151** [comment]
Now it works due to bypass

**Pull request closed without merging**